app/views/handlers/list_action.py

```diff
@@ -47,8 +47,27 @@ def handle_post(self):
             self.parent_location.block_id
         )
         self.questionnaire_store_updater.remove_answers(answer_ids_to_remove)
+        self.evaluate_and_update_section_status()
         self.questionnaire_store_updater.save()
 
+    def evaluate_and_update_section_status(self):
+        list_name = self.parent_block['for_list']
+        section_ids_to_filter_by = self._schema.get_section_ids_dependent_on_list(
+            list_name
+        )
+
+        section_ids_to_evaluate = self.questionnaire_store_updater.get_in_progress_and_completed_section_ids(
+            section_ids_to_filter_by
+        )
+
+        for section_id, list_item_id in section_ids_to_evaluate:
```

Comment:
`section_ids_to_evaluate` contains section keys, so should be named `section_keys_to_evaluate`